src/ui/components/Events/Event.tsx

```diff
@@ -1,13 +1,144 @@
+import {
+  ExecutionPoint,
+  Location,
+  Node,
+  Object as ProtocolObject,
+  MouseEvent as ReplayMouseEvent,
+  Value,
+} from "@replayio/protocol";
 import classNames from "classnames";
 import React, { KeyboardEvent, MouseEvent, ReactNode } from "react";
 
+import { selectLocation } from "devtools/client/debugger/src/actions/sources/select";
+import { getThreadContext } from "devtools/client/debugger/src/reducers/pause";
+import type { ThreadFront as TF } from "protocol/thread";
+import { createGenericCache } from "replay-next/src/suspense/createGenericCache";
+import { MAPPER as EVENTS_MAPPER, EventLog } from "replay-next/src/suspense/EventsCache";
+import { getFramesAsync } from "replay-next/src/suspense/FrameCache";
+import { getObjectWithPreviewHelper } from "replay-next/src/suspense/ObjectPreviews";
+import { getPauseIdAsync } from "replay-next/src/suspense/PauseCache";
+import { compareExecutionPoints } from "replay-next/src/utils/time";
+import { ReplayClientInterface } from "shared/client/types";
+import type { UIThunkAction } from "ui/actions";
+import {
+  FunctionWithPreview,
+  REACT_16_EVENT_LISTENER_PROP_KEY,
+  REACT_17_18_EVENT_LISTENER_PROP_KEY,
+  formatEventListener,
+} from "ui/actions/event-listeners";
 import useEventContextMenu from "ui/components/Events/useEventContextMenu";
+import { getPreferredLocation, getSourceDetailsEntities } from "ui/reducers/sources";
+import { useAppDispatch } from "ui/setup/hooks";
+import { UIState } from "ui/state";
 import { ReplayEvent } from "ui/state/app";
 import { getFormattedTime } from "ui/utils/timeline";
 
 import MaterialIcon from "../shared/MaterialIcon";
 import { getReplayEvent } from "./eventKinds";
 
+const { getValueAsync: getNextClickEventAsync } = createGenericCache<
+  [replayClient: ReplayClientInterface, point: ExecutionPoint, time: number],
+  EventLog | undefined
+>(
+  "clickEventCache",
+  async (replayClient, point, time) => {
+    // Actual browser click events get recorded a fraction later then the
+    // "mouse events" used by the sidebar.
+    // Look for the next click event within a short timeframe after the "mouse event".
+    // Yes, this is hacky, but it does seem pretty consistent.
+    const arbitraryEndTime = time + 200;
```

Comment:
This makes me nervous. I wonder how many times we may get false positives from this? How does this work for double clicks?

cc @jasonLaster @bhackett1024  in case they have more context about edge cases here.

Reply:
I don't know how the runtime specifically determines which events it records for the `Session.findMouseEvents` protocol request.  I think @Domiii recently implemented that for Chromium and might have more context on how that works internally.

That said, I checked this across several recordings and am consistently seeing _some_ `"click"` event listener entry show up very shortly after the timestamp for the `findMouseEvents` entry.  

Like I said, it's not ideal, but it's a sufficiently good heuristic for now.

Reply:
@markerikson I am surprised that this works, since `getPointNearTime` throws an error if there is no active region at exactly that point in time :smile:

If it does not error out, the `runAnalysis` part should probably work just fine on gecko.
However, on chromium, event names are a total mess, so we will have to talk, or I have to just write you a function that you can use to get the right events to query for.
Please keep that in mind when my `events` PR finally hopefully lands before/by next week :eyes:

Reply:
Hmm. Yeah, I've only actually tested on this relatively small recordings, and both made with Firefox.  Those are good points!  Let me think about this a bit more.

Reply:
@markerikson It would theoretically make sense for the mouse click event to be associated with its "raw event" somehow. The "raw event", as you might have already figured out, records both, begin and end points, so as to allow searching for the first user frame between the two.

Reply:
on the flip side: 

- why do we even _have_ `Session.findMouseEvents`?  What purpose does that API server?
- Why are these events labeled as "Click" in the UI, but with a type of `"mousedown"`?
- Why does there seem to be a different number of results for those `findMouseEvents` entries than there are for "real" `"click"` or `"mousedown"` events?

Reply:
@markerikson 
1. The three `Session.find*Events` calls are probably from an earlier time, and were thought of and added over time (I am guessing). The data points collected for them have more information attached to them. However, they might be somewhat legacy, since we are likely able to use the (very different) data we use for the `Debugger.find*Event*Count*` calls to get mostly the same data that we desire from `Session.find*Events`. Cleaning that up and reasoning about the different user scenarios, is probably going to be a low priority for now.
2. I am not sure, but the entire mixing-and-matching algorithm generally gets a bit hacky. The way it works in `gecko` is also different from `chromium`, and from what I saw, they both have different types of inaccuracies.
3. The `findMouseEvents` data points are collected upon event dispatch inside the render engine, so they should always register any "click". The `Debugger` events, on the other hand, should only report on events that have a user-space JS listener associated with them (but possibly with false positives). Furthermore, the `Debugger` events are collected using mechanisms but ultimately come down to `gecko`'s and `chromium`'s "Event Listener Breakpoint" feature. In `gecko`, if you enable logging for all "Event Listener Breakpoints", you should see the same events as in Replay. If that is not the case, feel free to file a bug. (I don't think anyone has bothered to look into this yet.) I also detected some inconsistencies, e.g. [here](https://linear.app/replay/issue/RUN-1061#comment-a6859804). In `chromium` it might be even worse because their "Event Listener Breakpoints" are not as well written, not well maintained, and they also don't have the "log-only" feature that `gecko` has.